### PR TITLE
Add concurrency setting to codeql workflow

### PR DESCRIPTION
**Pull request by snnn**
### Description
Add concurrency setting to codeql workflow

### Motivation and Context
To save machine resource.

### CODE REVIEW

**Comment by justinchuby** [comment]
I think we need to set path for the python format workflow too

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -49,10 +49,15 @@ jobs:
         with:
           toolchain: stable
           components: rustfmt
+      - name: Update PATH
+        run: |
+          echo "$HOME/.local/bin" >> "$GITHUB_PATH"
```
Comment (snnn) [inline .github/workflows/lint.yml:54]: Actually, I would prefer to add the new dir to the front of PATH env, but I don't know how to do it.

Comment (justinchuby) [reply .github/workflows/lint.yml:54]: I think `export GITHUB_PATH="$HOME/.local/bin:$GITHUB_PATH"`

Comment (snnn) [reply .github/workflows/lint.yml:54]: I guess here $GITHUB_PATH is a file.

Comment (justinchuby) [reply .github/workflows/lint.yml:54]: I see it's a file

**Review by justinchuby — APPROVED** [review]

**Review by mszhanyi — APPROVED** [review]

**Pull request merged**